### ISSUE TITLE
VS 2022 Preview 4.0 Code Analysis Does Not Work

### ISSUE THREAD

**Issue by mavasani** [dotnet/roslyn-analyzers#5785]
Manually ported from https://developercommunity.visualstudio.com/t/VS-2022-Preview-40-Code-Analysis-Does-N/1631628

See the attached screenshot. Both of the SuppressMessage attributes are greyed-out with IDE0079: Remove unneccesary suppression. If I remove those attributes, no code-analysis warnings are displayed.

If I open the same project in VS 2019, the SuppressMessage attributes are required to avoid code-analysis warnings. Manually running code analysis for the solution has no effect.

Possibly-relevant csproj settings:
```xml
    <EnableNETAnalyzers>true</EnableNETAnalyzers>
    <AnalysisMode>AllEnabledByDefault</AnalysisMode>
```

![image](https://user-images.githubusercontent.com/10605811/148537401-1a2d2838-6eb8-486f-87cb-775686532bae.png)

**Comment by mavasani** [comment]
We need to investigate if the lack of CA1815 and CA1062 in latest package versions is a regression or an actual fix in the analyzer which means the suppression is no longer required. If latter, the bug should be closed as by design.
